### PR TITLE
Add Defunct PIDs page

### PR DESCRIPTION
**Pull request by StevenBarre**

### CODE REVIEW

**Review by tbaker1313 — APPROVED** [review]

**Review by IanKWatts — APPROVED** [review]

**Review by wmhutchison — APPROVED** [review]
👍

File: src/docs/platform-automation/defunct-pids.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+## Quick Fix
+
+The quick fix is to just delete the pod. That will cause all the extra processes to exit and restart the pod in a clean state. The pod will likely build up excess defunct processes again in the future, but this can be a quick fix to stop the alert emails until you have time to more fully address the issue.
+
+## Proper fix
+
+Traditionally in containers, the mantra is "one container, one process" but in recent times, containers will often have several processes running in them. They may also include readyness and liveness checks that use exec.
+
+The first process to start in a container is treated as PID 1 and all other processes are treated as children of that process. If that process is not expecting to be parent to other processes, then when those child processes exit in unexpected ways they may become zombie or defunct and need to be cleaned up.
```
Comment (Pilargit12) [inline src/docs/platform-automation/defunct-pids.md:34]:
```suggestion
The first process in a container is PID 1, and all others are its children. If PID 1 doesn't expect to parent other processes when those child processes exit in unexpected ways, they can become zombie or defunct, needing cleanup.
```

File: src/docs/platform-automation/defunct-pids.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+The proper fix is to ensure that the `ENTRYPOINT` of your container is a process that can handle the cleanup of zombie child processes, often called an init program. Some examples include [Tini](https://github.com/krallin/tini), [dumb-init](https://github.com/Yelp/dumb-init), and [s6-overlay](https://github.com/just-containers/s6-overlay). Having one of these init processes as your PID 1 that then runs your main container process will ensure that any unhandled defunct PIDs are cleaned up instead of hanging around causing issues for the nodes kernel.
```
Comment (Pilargit12) [inline src/docs/platform-automation/defunct-pids.md:36]:
```suggestion
The proper solution is to set your container's `ENTRYPOINT` to a process that handles zombie child processes cleanup, called an init program. Examples include [Tini](https://github.com/krallin/tini), [dumb-init](https://github.com/Yelp/dumb-init), and [s6-overlay](https://github.com/just-containers/s6-overlay). Using one of these init processes as PID 1 ensures that unhandled defunct PIDs are cleaned up, preventing issues for the node's kernel.
```

File: src/docs/platform-automation/defunct-pids.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Sometimes minor changes to the probes can also be done to prevent them from creating zombies as well.
```
Comment (Pilargit12) [inline src/docs/platform-automation/defunct-pids.md:38]:
```suggestion
Sometimes, minor changes to the probes can prevent them from creating zombie processes.
```

File: src/docs/platform-automation/defunct-pids.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+---
+---
+
+## Additional Reading
```
Comment (Pilargit12) [inline src/docs/platform-automation/defunct-pids.md:43]:
```suggestion
## Additional resources
```

File: src/docs/platform-automation/defunct-pids.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+- [Container Init Process](https://devopsdirective.com/posts/2023/06/container-init-process/)
```
Comment (Pilargit12) [inline src/docs/platform-automation/defunct-pids.md:45]:
```suggestion
- [Container Init process](https://devopsdirective.com/posts/2023/06/container-init-process/)
```

File: src/docs/platform-automation/defunct-pids.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+- [bitnami/mongodb fix PR](https://github.com/bitnami/charts/pull/23390/files)
+- [bitnami/redis-cluster fix PR](https://github.com/bitnami/charts/pull/5335/files)
+- [Kubernetes Feature Request to add init process to all containers](https://github.com/kubernetes/kubernetes/issues/84210)
```
Comment (Pilargit12) [inline src/docs/platform-automation/defunct-pids.md:48]:
```suggestion
- [Kubernetes feature request to add init process to all containers](https://github.com/kubernetes/kubernetes/issues/84210)
```

File: mkdocs.yml [hunk 7]
```diff
@@ -66,6 +66,7 @@ nav:
   - AlertManager: docs/platform-automation/alertmanager.md
   - Pruning: docs/platform-automation/pruning.md
   - Kyverno cluster policies: docs/platform-automation/kyverno.md
+  - Defunct PIDs: docs/platform-automation/defunct-pids.md
```
Comment (Pilargit12) [inline mkdocs.yml:69]:
```suggestion
  - Defunct Process IDs (PIDs): docs/platform-automation/defunct-pids.md
```

File: src/docs/platform-automation/defunct-pids.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
+---
+title: Defunct PIDs
```
Comment (Pilargit12) [inline src/docs/platform-automation/defunct-pids.md:2]:
```suggestion
title: Defunct Process IDs (PIDs)
```

**Review by Pilargit12 — CHANGES_REQUESTED** [review]
Minor changes and we are missing an quick introduction to the subject as well

**Review by Pilargit12 — APPROVED** [review]

**Comment by Pilargit12** [comment]
Hi @StevenBarre if you are happy, please feel free to merge to main, thank you!

**Comment by StevenBarre** [comment]
Thanks, still working on the actual code part. Will merge once its ready to launch.

**Pull request merged**